src/store/modules/product/actions.ts

```diff
@@ -46,6 +46,40 @@ const actions: ActionTree<ProductState, RootState> = {
     // TODO Handle specific error
     return resp;
   },
+  async fetchProductDetails({ commit, state }, { productIds }) {
+    const cachedProductIds = Object.keys(state.cached);
+    const productIdFilter = productIds.reduce((filter: string, productId: any) => {
+      if(cachedProductIds.includes(productId)) {
+        return filter;
+      }
+      else {
+        if(filter !== '') filter += ' OR '
+        return filter += productId;
+      }
+    }, '');
+    if (productIdFilter === '') return;
+    const resp = await ProductService.fetchProducts({
+      "filters": ['productId: (' + productIdFilter + ')'],
+      "viewSize": productIds.length
```

Comment:
Instead make it `productIdFilter.length`, as in productIds we might be getting 20 ids but some of them may be in cache and we won't send them with the API

Reply:
Implemented code, Sir.